### PR TITLE
Remove deprecated linear algebra related functions

### PR DESCRIPTION
**Pull request by IvanYashchuk**
So the time has come to finally remove a few of the deprecated functions.

List of removed functions (split this PR into separate ones):
- torch.chain_matmul https://github.com/pytorch/pytorch/pull/70978
- torch.cholesky https://github.com/pytorch/pytorch/pull/70979
- torch.eig https://github.com/pytorch/pytorch/pull/70982
- torch.lstsq https://github.com/pytorch/pytorch/pull/70980
- torch.matrix_rank https://github.com/pytorch/pytorch/pull/70981
- torch.qr https://github.com/pytorch/pytorch/pull/70989
- torch.solve https://github.com/pytorch/pytorch/pull/70986
- torch.symeig https://github.com/pytorch/pytorch/pull/70988

Fixes to torch/xla need to be merged before this PR:

- [ ] https://github.com/pytorch/xla/pull/3272
- [ ] https://github.com/pytorch/xla/pull/3273
- [ ] https://github.com/pytorch/xla/pull/3274

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -7016,30 +6994,12 @@
...
-- func: symeig.e(Tensor self, bool eigenvectors=False, bool upper=True, *, Tensor(a!) e, Tensor(b!) V) -> (Tensor(a!) eigenvalues, Tensor(b!) eigenvectors)
-  dispatch:
-    CompositeExplicitAutograd: symeig_out
-
-- func: symeig(Tensor self, bool eigenvectors=False, bool upper=True) -> (Tensor eigenvalues, Tensor eigenvectors)
-  variants: method, function
-  dispatch:
-    CompositeExplicitAutograd: symeig
-
 - func: _symeig_helper(Tensor self, bool eigenvectors, bool upper) -> (Tensor, Tensor)
```
Comment (IvanYashchuk) [inline aten/src/ATen/native/native_functions.yaml:6997]: Remove `_symeig_helper`.

**Review by lezcano — COMMENTED** [review]
`eig_backward` should also be removed from `FunctionsManual.cpp` and the header file.

There's a minor clean-up that can be made in `eigh_backward` as well, where the `eigenvectors` argument was only used in `symeig`.

**Comment by IvanYashchuk** [comment]
I will split the PR into multiple smaller ones for an easier review process and easier reverting if needed.

**Pull request closed without merging**